.gitlab/e2e_test_junit_upload.yml

```diff
@@ -0,0 +1,15 @@
+source_test_junit_upload:
+  rules:
+    !reference [.on_main]
+  when: always
+  stage: e2e_test_junit_upload
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/datadog-ci-uploader$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:main"]
+  allow_failure: true
+  variables:
+    DD_ENV: ci
+  script:
+    - set +x
+    - export DATADOG_API_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.datadog_api_key_org2 --with-decryption --query "Parameter.Value" --out text)
+    - set -x
+    - for f in junit-new-e2e-*.tgz; do inv -e junit-upload --tgz-path $f; done
```

Comment:
> The main problem with this job is that it downloads artifacts from all previous stages (including all the package builds), making this very slow.

Isn’t this problem caused only by the fact that I initially forgot to set [the `dependencies` property](https://docs.gitlab.com/ee/ci/yaml/#dependencies) ?

If I do: https://github.com/DataDog/datadog-agent/blob/ab085362e9f9dcf6f7feac5cfa17912a40d4ca6b/.gitlab/e2e_test_junit_upload.yml#L11-L12
then this job should download only the artifacts of the `new-e2e-containers-main` job, which are only the junit reports: https://github.com/DataDog/datadog-agent/blob/ab085362e9f9dcf6f7feac5cfa17912a40d4ca6b/.gitlab/e2e.yml#L143-L147

If so, this concern should be addressed by https://github.com/DataDog/datadog-agent/pull/17950/commits/ab085362e9f9dcf6f7feac5cfa17912a40d4ca6b.

Reply:
This is another solution - the downside of it is that every time someone adds a new e2e test job, they need to update the `dependencies` of the `e2e_test_junit_upload` job, which is error-prone. If we think we can avoid this (through documentation, checks, etc.), we can go with this.

Reply:
Good point.
I’ve left some instructions at the location where someone would add new jobs so that it should be difficult to miss them: https://github.com/DataDog/datadog-agent/pull/17950/commits/9e27b1080f255293f932f7e637b1f243bc87d55f.